roxie/udplib/udptrs.cpp

```diff
@@ -164,17 +373,51 @@ class UdpReceiverEntry : public IUdpReceiverEntry
 
     unsigned sendData(const UdpPermitToSendMsg &permit, TokenBucket *bucket)
     {
+        if (permit.flowSeq != nextFlowSequence)
+        {
+            if (udpTraceLevel>1 || udpTraceFlow)
+            {
+                StringBuffer s;
+                DBGLOG("UdpFlow: ignoring out-of-date permit_to_send seq %" SEQF "u (expected %u) to node %s", permit.flowSeq, nextFlowSequence+0, permit.destNode.getTraceText(s).str());
+            }
+            return 0;
+        }
         requestExpiryTime = 0;
         unsigned maxPackets = permit.max_data;
         std::vector<DataBuffer *> toSend;
         unsigned totalSent = 0;
-        while (toSend.size() < maxPackets && packetsQueued.load(std::memory_order_relaxed))
+        unsigned resending = 0;
+        if (resendList)
+        {
+            resendList->noteRead(permit.seen, toSend, maxPackets, nextSendSequence.load(std::memory_order_relaxed));
+            resending = toSend.size();
```

Comment:
Should totalSent include the resent data?

Reply:
It should, and it does